plugins/extractit/icon_extractit/util/extractor.py

```diff
@@ -12,15 +12,22 @@
 from pdfminer.pdfparser import PDFSyntaxError
 
 
-def extract(provided_regex: str, provided_string: str, provided_file: str) -> list:
+def extract(provided_regex: str, provided_string: str, provided_file: str, keep_original_url=False) -> list:
     matches = []
     if provided_string:
-        provided_string = urllib.parse.unquote(provided_string)
-        matches = regex.findall(provided_regex, provided_string)
+        if keep_original_url:
+            matches = regex.findall(provided_regex, provided_string)  # regex finds both encoded and unencoded urls
+        else:
+            provided_string = urllib.parse.unquote(provided_string)
+            matches = regex.findall(provided_regex, provided_string)
     elif provided_file:
         try:
-            provided_file = urllib.parse.unquote(base64.b64decode(provided_file.encode("utf-8")).decode("utf-8"))
-            matches = regex.findall(provided_regex, provided_file)
+            if keep_original_url:
+                provided_file = base64.b64decode(provided_file.encode("utf-8")).decode("utf-8")
```

Comment:
Please make a constant variable `DEFAULT_ENCODING = "utf-8"` and use it like `decode(DEFAULT_ENCODING)`, you can place the variable at the top of this file after imports & before classes, or make a constants.py